File: apollo-cache/src/main/java/com/apollographql/apollo/internal/cache/normalized/CacheKeyBuilder.java
package com.apollographql.apollo.internal.cache.normalized;

import com.apollographql.apollo.api.Operation;
import com.apollographql.apollo.api.ResponseField;

import org.jetbrains.annotations.NotNull;

public interface CacheKeyBuilder {
  @NotNull String build(@NotNull ResponseField field, @NotNull Operation.Variables variables);
}

File: apollo-normalized-cache/src/main/java/com/apollographql/apollo/cache/normalized/internal/CacheKeyBuilder.kt
package com.apollographql.apollo.cache.normalized.internal;

import com.apollographql.apollo.api.Operation;
import com.apollographql.apollo.api.ResponseField;

import org.jetbrains.annotations.NotNull;

public interface CacheKeyBuilder {
  @NotNull String build(@NotNull ResponseField field, @NotNull Operation.Variables variables);
}
